system/modules/core/classes/DataContainer.php

```diff
@@ -511,7 +511,7 @@ protected function row($strPalette=null)
 
 				if ($objFile->isSvgImage || $objFile->height <= \Config::get('gdMaxImgHeight') && $objFile->width <= \Config::get('gdMaxImgWidth'))
 				{
-					if ($objFile->width > 699 || $objFile->height > 524)
+					if ($objFile->width > 699 || $objFile->height > 524 || !$objFile->width || !$objFile->height)
```

Comment:
If one of the two dimensions or both are missing we should resize the image, so IMO `!$objFile->width || !$objFile->height` is correct.